File: medium/golang/53-maximum-subarray.go
package golang

func maxSubArray(nums []int) int {
	maxSum := nums[0]
	curSum := 0
	for _, num := range nums {
		curSum += num
		maxSum = max(maxSum, curSum)
		curSum = max(curSum, 0)
	}
	return maxSum
}

File: medium/python/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        result = nums[0]
        sum = 0
        for i in range(0, len(nums)):
            sum += nums[i]
            result = max(result, sum)
            sum = max(sum, 0)
        return result
        